webapi/src/main/scala/org/knora/webapi/routing/admin/ProjectsRouteZ.scala

```diff
@@ -234,6 +246,31 @@ final case class ProjectsRouteZ(
       r                   <- projectsService.getProjectRestrictedViewSettings(shortcodeIdentifier)
     } yield Response.json(r.toJsValue.toString)
 
+  private def setProjectRestrictedViewSizeByIri(
+    iri: String,
+    body: Body,
+    user: UserADM
+  ): Task[Response] =
+    for {
+      iriDecoded <- RouteUtilZ.urlDecode(iri, s"Failed to URL decode IRI parameter $iri.")
+      id         <- IriIdentifier.fromString(iriDecoded).toZIO.mapError(e => BadRequestException(e.msg))
+      body       <- body.asString
+      payload    <- ZIO.fromEither(body.fromJson[ProjectSetRestrictedViewSizePayload]).mapError(BadRequestException(_))
+      size       <- ZIO.fromEither(RestrictedViewSize.make(payload.size)).mapError(BadRequestException(_))
+      response   <- projectsService.setProjectRestrictedViewSettings(id, user, size)
+    } yield Response.json(response.toJson)
+
+  private def setProjectRestrictedViewSizeByShortcode(
+    shortcode: String,
+    body: Body,
+    user: UserADM
+  ): Task[Response] = for {
+    id       <- ShortcodeIdentifier.fromString(shortcode).toZIO.mapError(e => BadRequestException(e.msg))
+    body     <- body.asString
```

Comment:
From this line to end of the function it is a duplicate from the function above. 
You could extract the shared code into a function"
```scala
private def setProjectRestrictedViewSizeByIri(
   id: ProjectIdentifier,
    body: Body,
    user: UserADM
  ): Task[Response]
```